### PR TITLE
feat: test for BattleStatus Enum

### LINKED ISSUE #92: 📝 [FEAT] Add Unit Tests for BattleStatus Enum and Conversions
### 🔍 Scope
The `BattleStatus` enum defines the various states of a battle and includes conversion implementations between `BattleStatus` and both `u8` and `felt252`. However, it currently lacks unit tests to ensure these conversions behave as expected.

This issue is focused on writing a complete set of unit tests for the enum and its impls to guarantee correctness, especially in the context of serialization and state handling.

### 🔄 How to Implement the Issue

- **Step 1:** Add tests for the following conversions:
  - `BattleStatus` to `felt252` using `Into<BattleStatus, felt252>`
  - `BattleStatus` to `u8` using `Into<BattleStatus, u8>`
  - `u8` to `BattleStatus` using `Into<u8, BattleStatus>`

- **Step 2:** Write assertions to verify:
  - Each enum variant returns the correct numeric representation.
  - Each numeric input maps back to the correct enum variant.
  - Fallback behavior (`_ => BattleStatus::None`) is properly triggered on invalid `u8` inputs.

- **Step 3:** Ensure the tests use meaningful descriptions and cover edge cases (e.g., `u8 = 255`).

### ✅ Acceptance Criteria
- [ ] Unit tests exist for each `Into` implementation.
- [ ] All valid conversions behave as expected.
- [ ] Invalid conversions return `BattleStatus::None`.
- [ ] Tests are concise, clear, and follow Dojo conventions.

### 📸 Additional Context
Enums like `BattleStatus` play a critical role in driving game logic and system behavior. Ensuring conversions are accurate is essential to prevent unintended bugs in the state machine of a battle system.

Enum source:  
https://github.com/AkatsukiLabs/DojoByExample/blob/main/backend/dojo_examples/combat_game/src/types/battle_status.cairo

### Contribution Guidelines
- Please ensure you read and follow the contribution guidelines in the project's [CONTRIBUTIONS GUIDELINES](https://github.com/AkatsukiLabs/DojoByExample/issues/3)

### PR DESCRIPTION
**Pull request by bitfalt**
# Pull Request Overview

## 📝 Summary
This PR provides unit testing for the `BattleStatus` enum.

### Related Issues
- Closes #92 
- 
### Type of Change
- [ ] 📝 Documentation (updates to README, docs, or comments)
- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] 👌 Enhancement (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)

### What's Changed
Add unit tests for the `BattleStatus` enum and add a `.tool-version` file for future contributors.

### Implementation Details
Validate each conversion and check if it behaves as expected. Checked for invalid input and the proposed edge case of 255.

### Test Coverage
- [x] Unit Tests
- [ ] Integration Tests
- [ ] Manual Testing

### Evidence
Screenshot 2025-04-26 at 18 07 06

### CODE REVIEW

File: backend/dojo_examples/combat_game/src/types/battle_status.cairo [hunk 1]
```diff
@@ -42,4 +42,107 @@ pub impl Intou8BattleStatus of Into<u8, BattleStatus> {
...
+        assert_eq!(battle_status, BattleStatus::None);
+    }
+
+    #[test]
+    fn test_into_battle_status_from_u8_255_edge_case() {
+        let battle_status_u8: u8 = 255;
+        let battle_status: BattleStatus = battle_status_u8.into();
+        assert_eq!(battle_status, BattleStatus::None);
+    }
 }
```
Comment (coxmars) [inline backend/dojo_examples/combat_game/src/types/battle_status.cairo:148]: @bitfalt can you add a new line at the end of the file please?

**Review by coxmars — APPROVED** [review]

File: backend/dojo_examples/combat_game/src/types/battle_status.cairo (hunk 1 above)
Comment (bitfalt) [reply backend/dojo_examples/combat_game/src/types/battle_status.cairo]: done!

**Review by jimenezz22 — APPROVED** [review]
excelent @bitfalt
